auth.go

```diff
@@ -440,6 +378,101 @@ func authenticate(
 	return &respd.Data, nil
 }
 
+func tryAuth(
+	ctx context.Context,
+	sc *snowflakeConn,
+	samlResponse []byte,
+	proofKey []byte,
+	sessionParameters map[string]interface{},
+	clientEnvironment authRequestClientEnvironment,
+	params *url.Values,
+	headers map[string]string,
+) (*authResponse, error) {
+	authRequest, err := buildAuthRequestData(sc, sessionParameters, clientEnvironment, proofKey, samlResponse)
+	if err != nil {
+		return nil, err
+	}
+	jsonBody, err := json.Marshal(authRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.WithContext(sc.ctx).Infof("PARAMS for Auth: %v, %v, %v, %v, %v, %v",
+		params, sc.rest.Protocol, sc.rest.Host, sc.rest.Port, sc.rest.LoginTimeout, sc.cfg.Authenticator.String())
+
+	respd, err := sc.rest.FuncPostAuth(ctx, sc.rest, params, headers, jsonBody, sc.rest.LoginTimeout)
+	if err != nil {
+		return nil, err
+	}
+	return respd, nil
+}
+
+func buildAuthRequestData(
+	sc *snowflakeConn,
+	sessionParameters map[string]interface{},
+	clientEnvironment authRequestClientEnvironment,
+	proofKey []byte,
+	samlResponse []byte,
+) (*authRequest, error) {
```

Comment:
do we really need to return a pointer here?

Reply:
Yes, otherwise `nil` can't be returned. This is also more lightweight.